azure/services/managedclusters/managedclusters.go

```diff
@@ -147,3 +157,75 @@ func (s *Service) Delete(ctx context.Context) error {
 func (s *Service) IsManaged(ctx context.Context) (bool, error) {
 	return true, nil
 }
+
+// ReconcileKubeconfig will reconcile admin kubeconfig and user kubeconfig.
+/*
+  Returns the admin kubeconfig and user kubeconfig
+  If aad is enabled a user kubeconfig will also get generated and stored in the secret <cluster-name>-kubeconfig-user
+  If we disable local accounts for aad clusters we do not have access to admin kubeconfig, hence we need to create
+  the admin kubeconfig by authenticating with the user credentials and retrieving the token for kubeconfig.
+  The token is used to create the admin kubeconfig.
+  The user needs to ensure to provide service principle with admin aad privileges.
+*/
+func (s *Service) ReconcileKubeconfig(ctx context.Context, managedClusterSpec azure.ResourceSpecGetter) (userKubeConfigData []byte, adminKubeConfigData []byte, err error) {
```

Comment:
Is it possible to add a unit test for this function? I think it can be useful here but if it requires too much mocking we can track this in a separate issue.

Reply:
let me try to do that.

Reply:
@willie-yao I am unable to figure out how to mock this and add more testing for this, actually a part of the testing is already covered in the below test-case. 

https://github.com/spectrocloud/cluster-api-provider-azure/blob/0abf40eced4dac96e0869c036df5cd243bfd3e49/azure/services/managedclusters/managedclusters_test.go#L105-#L142

What else do you think i can do to improve the coverage ?

Reply:
I think in this case, the test coverage is fine as-is since it is already covered in the test case listed above. Thanks for looking into it!